channel.py

```diff
@@ -377,6 +377,53 @@ def _payload(self) -> JsonType:
         return {"data": (b64encode(json_minify(payload).encode("utf8"))).decode("utf8")}
 
     async def send_watch(self) -> bool:
+        #-------------------------------------------------
+        # FIX 2024/5
+        #  |
+        #  v
+        try:
+            response: JsonType = await self._twitch.gql_request(        # Gets signature and value
+                GQL_OPERATIONS["PlaybackAccessToken"].with_variables({"login": self._login})
+                )
+        except MinerException as exc:
+            raise MinerException(f"Channel: {self._login}") from exc
+        signature: JsonType | None = response["data"]['streamPlaybackAccessToken']["signature"]
+        value: JsonType | None = response["data"]['streamPlaybackAccessToken']["value"]
+        if not signature or not value:
+            return None
+
+        RequestBroadcastQualitiesURL = f"https://usher.ttvnw.net/api/channel/hls/{self._login}.m3u8?sig={signature}&token={value}"
```

Comment:
事实上你不需要每次都请求  
```RequestBroadcastQualitiesURL = f"https://usher.ttvnw.net/api/channel/hls/{self._login}.m3u8?sig={signature}&token={value}"```
为每个频道记录它（例如放到 self._spade_url），
接着只需要每60秒一次，请求低码率的m3u8.MEDIA并访问其中的任一`ts`链接。

Reply:
这样每分钟的流量消耗大约是 100k，仍旧处于一个可用状态

Reply:
> 事实上你不需要每次都请求  
> ```RequestBroadcastQualitiesURL = f"https://usher.ttvnw.net/api/channel/hls/{self._login}.m3u8?sig={signature}&token={value}"```
> 为每个频道记录它（例如放到 self._spade_url），
> 接着只需要每60秒一次，请求低码率的m3u8.MEDIA并访问其中的任一`ts`链接。
> 
> 

The signature is differend each time, even on the same channel. I will try that once I get home tho.

Reply:
> > 事实上你不需要每次都请求
> > `RequestBroadcastQualitiesURL = f"https://usher.ttvnw.net/api/channel/hls/{self._login}.m3u8?sig={signature}&token={value}"`
> > 为每个频道记录它（例如放到 self._spade_url），
> > 接着只需要每60秒一次，请求低码率的m3u8.MEDIA并访问其中的任一`ts`链接。
> 
> The signature is differend each time, even on the same channel. I will try that once I get home tho.

这个过程大致如下，你可以验证一下，祝你好运
```
import time
import m3u8
import requests

# step 1, stub
def update_spade():
    channel = "anonymous"
    res = requests("https://gql.twitch.tv/gql", "PlaybackAccessToken")
    sig = res["signature"]
    token = res["value"]

    mediaUrl = m3u8.load(f"https://usher.ttvnw.net/api/channel/hls/{channel}.m3u8?allow_source=true&sig={sig}&token={token}")
    return mediaUrl.segments[0].uri

### step 2
def watch_video(url):
    playlist = m3u8.load(url)
    r = requests.get(playlist.segments[0].uri)
    print(len(r.text))


# change for yourself
spadeUrl  = 'https://video-weaver.lax03.hls.ttvnw.net/v1/playlist/???.m3u8'

while True:
    # update mediaUrl as spadeUrl
    if spadeUrl == None:
        spadeUrl = update_spade()

    watch_video(spadeUrl)
    time.sleep(59)
```

Reply:
You don't need a GET request, which then downloads the actual video file, HEAD is enough. But what you propose would eequire a rewrite of the entire watch loop, which would take a bit of time, while people are waiting for the fix. Would put this in the efficiency issue, where I'm investigating the frequency.

Reply:
It's a good idea to use HEAD instead of GET, if available. This usually depends on server-side routing. I will try it.Also, I haven't encountered the frequency problem you mentioned.

Reply:
What's with the non-english non-technical comments? I too speak 7 languages but I do not bother other people with them.

Reply:
> What's with the non-english non-technical comments? I too speak 7 languages but I do not bother other people with them.

I'm sorry for the trouble, I only speak one language and my English is very poor. ignore me.